### PR TITLE
chore(upgrade): v1.0.0 to polkadot-sdk v1.1.0

### PR DESCRIPTION
**Pull request by enddynayn**
The goal of this PR is to upgrade from v1.0.0 to v1.1.0.

Benchmarks updated for custom pallets and overhead.

References
- [Polkadot Forum](https://forum.polkadot.network/t/polkadot-release-analysis-v1-1-0/3988)
- [Release Notes](https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.1.0)

Closes Issue #1687

### CODE REVIEW

File: runtime/frequency/src/lib.rs [hunk 1]
```diff
@@ -788,6 +794,18 @@ impl pallet_frequency_tx_payment::Config for Runtime {
 	type MaximumCapacityBatchLength = MaximumCapacityBatchLength;
 }
 
+#[cfg(any(not(feature = "frequency-no-relay"), feature = "frequency-lint-check"))]
```
Comment (enddynayn) [inline runtime/frequency/src/lib.rs:797]: revisit for last review

Comment (JoeCap08055) [reply runtime/frequency/src/lib.rs:797]: Some explanation here might be nice. Perhaps I'm reading it backwards, but it looks like these configs would be included for linting OR if we're building with the relay chain enabled... based on these values, I would think we would include them only for an instant-seal node... am I misreading something?

Comment (wilwade) [reply runtime/frequency/src/lib.rs:797]: I believe this is referring to this entire new section about preping for async backing

**Review by wilwade — COMMENTED** [review]
Looks like a straightforward upgrade outside of the Async backing updates.

Few questions, but nothing blocking as far as I can see.

- Reviewed Code
- Ran instant sealing locally

File: runtime/frequency/src/lib.rs (hunk 1 above)
Comment (mattheworris) [reply runtime/frequency/src/lib.rs:797]: These values are for setting up Async Backing. They are the values from the parachain-template.

File: pallets/schemas/Cargo.toml [hunk 2]
```diff
@@ -14,32 +14,33 @@ targets = ["x86_64-unknown-linux-gnu"]
...
 ] }
 serde_json = { version = "1.0.86", default-features = false, features = [
   "alloc",
 ] }
-smallvec = "1.10.0"
-sp-core = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v1.0.0" }
-sp-io = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v1.0.0" }
-sp-runtime = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v1.0.0" }
-sp-std = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v1.0.0" }
+smallvec = "1.11.0"
```
[the remark this answers is not in the document]
Comment (wilwade) [reply pallets/schemas/Cargo.toml:30]: Nothing super important like security updates in 1.11.1, so I think it is fine to keep as is then.

File: runtime/frequency/src/lib.rs [hunk 3]
```diff
@@ -228,14 +228,20 @@ pub type Executive = frame_executive::Executive<
 /// to even the core data structures.
 pub mod opaque {
 	use super::*;
+	use sp_runtime::{
```
Comment (shannonwells) [inline runtime/frequency/src/lib.rs:231]: Non-blocking: Do we have a convention for `use` statements in files and modules?  It seems kind of arbitrary.

**Review by shannonwells — COMMENTED** [review]
I've looked through all these files. Will approve once CI is passing, if needed.

File: runtime/frequency/src/lib.rs (hunk 3 above)
Comment (enddynayn) [reply runtime/frequency/src/lib.rs:231]: This follows templates Polkadot template. We tried to match it as much as possible.

File: pallets/msa/src/weights.rs [hunk 4]
```diff
@@ -96,17 +96,18 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	/// Storage: `Msa::DelegatorAndProviderToDelegation` (r:1 w:1)
 	/// Proof: `Msa::DelegatorAndProviderToDelegation` (`max_values`: None, `max_size`: Some(217), added: 2692, mode: `MaxEncodedLen`)
 	/// Storage: `Schemas::CurrentSchemaIdentifierMaximum` (r:1 w:0)
-	/// Proof: `Schemas::CurrentSchemaIdentifierMaximum` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Proof: `Schemas::CurrentSchemaIdentifierMaximum` (`max_values`: Some(1), `max_size`: Some(2), added: 497, mode: `MaxEncodedLen`)
```
Comment (enddynayn) [inline pallets/msa/src/weights.rs:99]: how did we get MaxEncodedLen here?

**Review by wilwade — APPROVED** [review]
Looks good to me.

Might be worth cleaning up the cargo deny warnings, but not a blocker.

- Ran various local tests
- Ran several versions

**Review by aramikm — COMMENTED** [review]
Overall looks good a couple of questions and concerns

- Since try-runtime is separated into a different binary I think out try-runtime script would fail but it was failing before so not a blocker
- concern: Are we sure if there is no internal storage migrations for the frame-pallets that we use?

File: pallets/stateful-storage/src/weights.rs [hunk 5]
```diff
@@ -72,16 +72,16 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
...
 		// Proof Size summary in bytes:
 		//  Measured:  `33370`
-		//  Estimated: `45745`
-		// Minimum execution time: 96_061_000 picoseconds.
-		Weight::from_parts(94_653_993, 45745)
-			// Standard Error: 393
-			.saturating_add(Weight::from_parts(7_470, 0).saturating_mul(s.into()))
+		//  Estimated: `77893`
+		// Minimum execution time: 99_564_000 picoseconds.
+		Weight::from_parts(97_543_524, 77893)
```
[the remark this answers is not in the document]
Comment (wilwade) [reply pallets/stateful-storage/src/weights.rs:77]: It is fine for now. We can get yours in the same release.

File: runtime/frequency/src/lib.rs [hunk 6]
```diff
@@ -800,9 +818,20 @@ impl cumulus_pallet_parachain_system::Config for Runtime {
 	type OutboundXcmpMessageSource = ();
 	type XcmpMessageHandler = ();
 	type ReservedXcmpWeight = ();
-	type CheckAssociatedRelayNumber = RelayNumberStrictlyIncreases;
+	type CheckAssociatedRelayNumber = RelayNumberMonotonicallyIncreases;
+	// TODO: Remove this when Async Backing is activated and the feature is set for cumulus-pallet-parachain-system
+	#[cfg(feature = "parameterized-consensus-hook")]
```
[the remark this answers is not in the document]
Comment (enddynayn) [reply runtime/frequency/src/lib.rs:823]: We can probably do that when we are ready to enable the feature.

File: node/service/src/service.rs [hunk 7]
```diff
@@ -317,9 +325,27 @@ async fn start_node_impl(
 		.overseer_handle()
 		.map_err(|e| sc_service::Error::Application(Box::new(e)))?;
 
+	start_relay_chain_tasks(StartRelayChainTasksParams {
+		client: client.clone(),
+		announce_block: announce_block.clone(),
+		para_id,
+		relay_chain_interface: relay_chain_interface.clone(),
+		task_manager: &mut task_manager,
+		da_recovery_profile: if validator {
```
[the remark this answers is not in the document]
Comment (mattheworris) [reply node/service/src/service.rs:334]: This code comes from the parachain template here: https://github.com/paritytech/polkadot-sdk/blob/495d24d730bb9c2205a5305f383627eabda4a323/cumulus/parachain-template/node/src/service.rs#L291

**Comment by enddynayn** [comment]
> Overall looks good a couple of questions and concerns
> 
> * Since try-runtime is separated into a different binary I think our try-runtime script would fail but it was failing before so not a blocker
> * concern: Are we sure if there is no internal storage migrations for the frame-pallets that we use?

I did not see any migrations but perhaps I missed one @mattheworris?

File: node/service/src/service.rs (hunk 7 above)
[the remark this answers is not in the document]
Comment (enddynayn) [reply node/service/src/service.rs:334]: https://paritytech.github.io/polkadot-sdk/master/cumulus_client_service/enum.DARecoveryProfile.html

Additionally, it seems these are the only options it takes.

**Comment by mattheworris** [comment]
> > Overall looks good a couple of questions and concerns
> > 
> > * Since try-runtime is separated into a different binary I think our try-runtime script would fail but it was failing before so not a blocker
> > * concern: Are we sure if there is no internal storage migrations for the frame-pallets that we use?
> 
> I did not see any migrations but perhaps I missed one @mattheworris?

There are no migrations called out in the release notes for v1.1.0.

**Review by aramikm — APPROVED** [review]
Great Job! Good to go!

**Review by saraswatpuneet — COMMENTED** [review]
I have reviewed the changes and looks good, however wont be able to approve since I havent run full tests

Otherwise good to go

**Comment by enddynayn** [comment]
> I have reviewed the changes and looks good, however wont be able to approve since I havent run full tests
> 
> Otherwise good to go

Please take the time to run the full test that you need to give it your approval.

File: runtime/common/src/weights/extrinsic_weights.rs [hunk 8]
```diff
@@ -43,17 +43,17 @@ parameter_types! {
...
 	///
 	/// Stats nanoseconds:
-	///   Min, Max: 98_988, 105_759
-	///   Average:  100_451
-	///   Median:   100_304
-	///   Std-Dev:  1082.34
+	///   Min, Max: 100_113, 110_777
+	///   Average:  101_295
+	///   Median:   101_040
+	///   Std-Dev:  1335.33
```
Comment (shannonwells) [inline runtime/common/src/weights/extrinsic_weights.rs:49]: Just noting that this is a jump of ~30% in std-dev which means the weight calculations are varying more widely in this benchmark run.

File: runtime/common/Cargo.toml [hunk 9]
```diff
@@ -12,39 +12,39 @@ version = "0.0.0"
 targets = ["x86_64-unknown-linux-gnu"]
 
 [build-dependencies]
-substrate-wasm-builder = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v1.0.0" }
+substrate-wasm-builder = { git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.1.0", optional = true }
```
[the remark this answers is not in the document]
Comment (mattheworris) [reply runtime/common/Cargo.toml:15]: It appears that `[build-dependencies]` may not recognize `optional = true` and in testing, removing this dependency has no effects. I am creating a new issue to remove this in a follow-on PR for cleaning up a few minor fixes.

File: node/service/src/service.rs (hunk 7 above)
[the remark this answers is not in the document]
Comment (JoeCap08055) [reply node/service/src/service.rs:334]: Should we seek more understanding of these values and what they do, or put our faith in the correctness of Parity's template code?

[the remark this answers is not in the document]
Comment (enddynayn) [reply node/service/src/service.rs:334]: @JoeCap08055 Please double check this but this seems correct to me. Full nodes simply follow the chain they do no create blocks. If you are a validator you should be an authoring node. This leaves us with the collator option.

**Review by shannonwells — APPROVED** [review]
Good work, ship it!

**Review by JoeCap08055 — APPROVED** [review]

**Pull request merged**